src/scenarios/shared/startup.py

```diff
@@ -101,11 +102,24 @@ def runtests(self, traits: TestTraits):
         if traits.innerloopcommandargs:
             startup_args.extend(['--inner-loop-command-args', traits.innerloopcommandargs])
             
-        RunCommand(startup_args, verbose=True).run()
+        upload_container = UPLOAD_CONTAINER
 
+        try:
+            RunCommand(startup_args, verbose=True).run()
+        except CalledProcessError:
+            upload_container = 'failedresults'
+            reportjson = os.path.join(
+                TRACEDIR,
+                'FailureReporter', 
+                'failure-report.json')
+            cmdline = [
+                'FailureReporting.exe', reportjson
+            ]
+            RunCommand(cmdline, verbose=True).run(
+                os.path.join(helixpayload(), 'FailureReporter'))
```

Comment:
After we create the path to the executable, we should check to make sure it exists, and if it doesn't we should throw here so we don't upload something that we do not intend.